interval/lift.go

```diff
@@ -11,3 +17,95 @@ type Lift interface {
 	WriteToFileHandle(*fileio.EasyWriter)
 	UpdateLift(string, int, int)
 }
+
+
+func GoReadToLiftChan(inputFile string) <-chan Lift {
+	answer := make(chan Lift, 1000)
+	go ReadToLiftChan(inputFile, answer)
+	return answer
+}
+
+func ReadToLiftChan(inputFile string, send chan<- Lift) {
+	// How the file is read is dependent on the file extension
+	filetype := path.Ext(inputFile)
+
+	if filetype == ".gz" {
```

Comment:
This is cool.  Seems like it could be useful other places.  Maybe in the future we could have a standard function in fileio that takes a filename and returns the type?

Reply:
yes I think this should be reworked in the future for a more general solution that switches on type, as you said